pyvisa_py/testsuite/keysight_assisted_tests/test_resource_manager.py

```diff
@@ -14,9 +19,15 @@
 class TestPyResourceManager(BaseTestResourceManager):
     """ """
 
-    test_list_resource = pytest.mark.xfail(
-        copy_func(BaseTestResourceManager.test_list_resource)
-    )
+    def test_list_resource(self):
+        """Test listing the available resources.
+        The bot supports only TCPIP and of those resources we expect to be able
+        to list only INSTR resources not SOCKET.
+        """
+        # Default settings
+        resources = self.rm.list_resources()
+        for v in (v for v in RESOURCE_ADDRESSES.values() if v.endswith("INSTR")):
+            assert str(ResourceName.from_string(v)) in resources
```

Comment:
Surprisingly for an instrument on the loop back I get the default gateway 192.168.1.56 which is annoying but I have no idea how to fix it.

Reply:
Hmmm, I'll need to see if I can reproduce this on my side. It might be a limitation of UDP broadcasts -- a device cannot respond to its own broadcast (so if a machine has a virtual instrument hosted on the same IP as the discovery process, it can't see itself). I'll look into this.

Reply:
The result is not wrong (I can actually address the device through that IP address) just unexpected. But if you figure out how to fix it all the best.